tools/buildtools/ambertools/parmconv.xml

```diff
@@ -0,0 +1,197 @@
+    <requirement type="package" version="2019.1">gromacs</requirement>
+    <requirement type="package" version="2.11.1">jinja2</requirement>
+  </expand>
+  <command detect_errors="exit_code">
+    <![CDATA[
+   bash '$bash_script' \$CONDA_PREFIX  &&
+   python '$templating_script' '$inputs' &&
+   export AMBERHOME=\$CONDA_PREFIX &&
+   parmed -i ligand.script -O &&
+   parmed -i receptor.script -O &&
+   parmed -i complex.script -O &&
+   parmed -i solvatedcomplex.script -O
+
+      ]]>
+  </command>
+  <configfiles>
+    <inputs name="inputs"/>
+    <configfile name="bash_script">
+      <![CDATA[
+      for i in "$1/share/gromacs/top/*.ff"; do ln -s \$i -f; done
+    ]]>
+    </configfile>
+    <configfile name="templating_script">
+      <![CDATA[
+
+import os
+import sys
+import json
+
+from jinja2 import Environment, FileSystemLoader
+input_json_path = sys.argv[1]
+params = json.load(open(input_json_path, "r"))
+currentpath = "$__tool_directory__"  # should work generally
+template_environment = Environment(loader=FileSystemLoader(currentpath),lstrip_blocks=True, trim_blocks=True)
+template = template_environment.get_template('template_parmconv.j2')
+print(params)
+params['fmt'] = '$param_inputs.fmt'
+#if str($param_inputs.fmt)  == 'AMBER':
+params["top_in"] = '$param_inputs.top_in'
+#else:
+params["top_in"] = '$param_inputs.top_in'
+params["str_in"] = '$param_inputs.str_in'
+#end if
+params['prmtop_ligand'] =   '$prmtop_ligand'
+params['prmtop_receptor'] =  '$prmtop_receptor'
+params['prmtop_complex'] =   '$prmtop_complex'
+params['prmtop_solvatedcomplex'] = '$prmtop_solvatedcomplex'
+print(params)
+
+def run_template(params=params, system="ligand"):
+    localparams=params.copy() # shallow copy ok for simple variables
+    localparams['stripmask']=localparams['stripmask_'+system]
+    localparams['prmtop_out']=localparams['prmtop_'+system]
+    print(localparams)
+    with open(system+'.script','w+') as f:
+        f.write(template.render(localparams))
+
+systems = ['ligand', 'receptor', 'complex', 'solvatedcomplex']
+for system in systems:
+    run_template(system=system)
+
+]]>
+    </configfile>
+  </configfiles>
+  <inputs>
+    <conditional name="param_inputs">
+      <param name="fmt" type="select" label="Force Field format">
+        <option selected="True" value="AMBER">AMBER</option>
+        <option value="GROMACS">GROMACS</option>
+      </param>
+      <when value="AMBER">
+        <param name="top_in" type="data" label="Input Topology file" format="txt"/>
+      </when>
+      <when value="GROMACS">
+        <param name="top_in" type="data" label="Input Topology file" format="top"/>
+        <param name="str_in" type="data" label="Input Structure file" format="gro"/>
+      </when>
+    </conditional>
+    <param name="stripmask_ligand" type="text" label="Ligand selection" value="!:UNL" help="Define a valid AMBER stripmask that will select only the ligand"></param>
```

Comment:
can we add a validator to all text fields?

Reply:
yes. Can contain !,): or any ligand name so it can be pretty ambiguous. 
Any suggestions? @simonbray

Reply:
not really - probably just `!,():` is best

Reply:
Some examples here https://amberhub.chpc.utah.edu/atom-mask-selection-syntax/
Masks can support @,* etc. 
I tried this example (:11@CD<:5.5)&:Na+ but the job fails and the parameters looks like this '(:11__at__CD__lt__:5.5)X:Na+'
Any ideas?

Reply:
found a way to map in the chemical toolbox :smile:

Reply:
does this help? https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-inputs-param-sanitizer

Reply:
I don't have a good idea how to write a validator unfortunately

Reply:
Thanks @simonbray I have been diving into the docs

Reply:
I think string.printable will work OK. '&' turns into X so I mapped it properly and included a complex selection usecase.